internal/unionstore/memdb_art.go

```diff
@@ -154,203 +153,49 @@ func (db *artDBWithContext) IterReverse(upper, lower []byte) (Iterator, error) {
 	return db.ART.IterReverse(upper, lower)
 }
 
-func (db *artDBWithContext) ForEachInSnapshotRange(lower []byte, upper []byte, f func(k, v []byte) (stop bool, err error), reverse bool) error {
-	db.RLock()
-	defer db.RUnlock()
-	var iter Iterator
-	if reverse {
-		iter = db.SnapshotIterReverse(upper, lower)
-	} else {
-		iter = db.SnapshotIter(lower, upper)
-	}
-	defer iter.Close()
-	for iter.Valid() {
-		stop, err := f(iter.Key(), iter.Value())
-		if err != nil {
-			return err
-		}
-		err = iter.Next()
-		if err != nil {
-			return err
-		}
-		if stop {
-			break
-		}
-	}
-	return nil
+// SnapshotGetter implements the Getter interface, by wrapping GetSnapshot.
+func (db *artDBWithContext) SnapshotGetter() Getter {
+	return db.ART.GetSnapshot()
 }
 
 // SnapshotIter returns an Iterator for a snapshot of MemBuffer.
 func (db *artDBWithContext) SnapshotIter(lower, upper []byte) Iterator {
```

Comment:
Does this method still need to be public? 
Will we replace original calls to `SnapshotIter` with `GetSnapshot().ForEachInSnapshotRange()` and `GetSnapshot().BatchedSnapshotIter()`?

Reply:
This interface is no more needed for tidb, but leave it for keeping the interface compatible.

Reply:
Maybe add `Deprecated:` to mark them and avoid new usages?